core/lib/external_price_api/src/coingecko_api.rs

```diff
@@ -39,18 +52,27 @@ impl CoinGeckoPriceAPIClient {
             reqwest::Client::new()
         };
 
-        let base_url = config
-            .base_url
-            .unwrap_or(DEFAULT_COINGECKO_API_URL.to_string());
+        let erc20 = config.forced_ethereum_erc20_address;
+        let forced_ethereum_erc20_address: Option<Address> = match erc20 {
+            Some(erc20) => match Address::from_str(erc20.as_str()) {
+                Ok(address) => Some(address),
+                Err(_) => None,
+            },
+            None => None,
+        };
 
         Self {
             base_url: Url::parse(&base_url).expect("Failed to parse CoinGecko URL"),
             client,
+            forced_ethereum_erc20_address,
         }
     }
 
     async fn get_token_price_by_address(&self, address: Address) -> anyhow::Result<f64> {
-        let address_str = address_to_string(&address);
+        let address_str: String = match self.forced_ethereum_erc20_address {
+            Some(erc20) => address_to_string(&erc20),
+            None => address_to_string(&address),
+        };
```

Comment:
Do I get it correctly that idea here is to override token address with some token that is present on mainnet?

Reply:
Yes, to accurately reflect the complete workflow in a testing environment.